app/workers/application_worker.rb

```diff
@@ -0,0 +1,9 @@
+class ApplicationWorker
+  include Sidekiq::Worker
+
+private
+
+  def run_with_advisory_lock(model, unique_ref, &block)
```

Comment:
This method is only used by the `PostcodesCollectionWorker`, so I wouldn't bother using inheritance here, I'd just move this method directly into that class, and delete this class.